all_values.sort() would cause compilation error but np.sort(all_values) would not

The error-free version is:

```
import numpy as np

#pythran export _pythran_cdf_distance(int, float[:], float[:])
#pythran export _pythran_cdf_distance(int, float[:], float[:], None, None)
#pythran export _pythran_cdf_distance(int, float[:], float[:], float[:], float[:])
def _pythran_cdf_distance(p, u_values, v_values, u_weights=None, v_weights=None):
    u_sorter = np.argsort(u_values)
    v_sorter = np.argsort(v_values)

    all_values = np.concatenate((u_values, v_values))
    all_values = np.sort(all_values, kind='mergesort')

    # Compute the differences between pairs of successive values of u and v.
    deltas = np.diff(all_values)

    # Get the respective positions of the values of u and v among the values of
    # both distributions.
    u_cdf_indices = np.searchsorted(u_values[u_sorter],all_values[:-1],'right')
    v_cdf_indices = np.searchsorted(v_values[v_sorter],all_values[:-1],'right')

    # Calculate the CDFs of u and v using their weights, if specified.
    if u_weights is None:
        u_cdf = u_cdf_indices / u_values.size
    else:
        u_sorted_cumweights = np.concatenate(([0],
                                              np.cumsum(u_weights[u_sorter])))
        u_cdf = u_sorted_cumweights[u_cdf_indices] / u_sorted_cumweights[-1]

    if v_weights is None:
        v_cdf = v_cdf_indices / v_values.size
    else:
        v_sorted_cumweights = np.concatenate(([0],
                                              np.cumsum(v_weights[v_sorter])))
        v_cdf = v_sorted_cumweights[v_cdf_indices] / v_sorted_cumweights[-1]

    # Compute the value of the integral based on the CDFs.
    # If p = 1 or p = 2, we avoid using np.power, which introduces an overhead
    # of about 15%.
    if p == 1:
        return np.sum(np.multiply(np.abs(u_cdf - v_cdf), deltas))
    if p == 2:
        return np.sqrt(np.sum(np.multiply(np.square(u_cdf - v_cdf), deltas)))
    return np.power(np.sum(np.multiply(np.power(np.abs(u_cdf - v_cdf), p),
                                       deltas)), 1/p)
```

If I modify `all_values = np.sort(all_values, kind='mergesort')` to `all_values.sort(kind='mergesort')`, the error would be:

```
CRITICAL: I am in trouble. Your input file does not seem to match Pythran's constraints...
_cdf_distance_1d.py:11:4 error: function uses an unknown (or unsupported) keyword argument `kind`
----
    all_values.sort(kind='mergesort')
    ^~~~ (o_0)
----
```

If I modify `all_values = np.sort(all_values, kind='mergesort')` to `all_values.sort()`, the error would be:

```
clang-11: warning: -Wl,-export_dynamic: 'linker' input unused [-Wunused-command-line-argument]
WARNING: Compilation error, trying hard to find its origin...
WARNING: Nop, I'm going to flood you with C++ errors!
CRITICAL: Cover me Jack. Jack? Jaaaaack!!!!
E: error: Command "x86_64-apple-darwin13.4.0-clang++ -fno-strict-aliasing -Wsign-compare -Wunreachable-code -DNDEBUG -fwrapv -O3 -Wall -march=core2 -mtune=haswell -mssse3 -ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O3 -pipe -fdebug-prefix-map=${SRC_DIR}=/usr/local/src/conda/${PKG_NAME}-${PKG_VERSION} -fdebug-prefix-map=/Users/charlotte/anaconda3/envs/scipydev=/usr/local/src/conda-prefix -flto -Wl,-export_dynamic -march=core2 -mtune=haswell -mssse3 -ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O3 -march=core2 -mtune=haswell -mssse3 -ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -pipe -isystem /Users/charlotte/anaconda3/envs/scipydev/include -D_FORTIFY_SOURCE=2 -isystem /Users/charlotte/anaconda3/envs/scipydev/include -DENABLE_PYTHON_MODULE -D__PYTHRAN__=3 -DPYTHRAN_BLAS_BLAS -I/Users/charlotte/anaconda3/envs/scipydev/lib/python3.8/site-packages/pythran -I/Users/charlotte/anaconda3/envs/scipydev/lib/python3.8/site-packages/numpy/core/include -I/Users/charlotte/anaconda3/envs/scipydev/include -I/Users/charlotte/anaconda3/envs/scipydev/include/python3.8 -c /var/folders/j_/zm2rmlh549733gxckfkzj9000000gn/T/tmp4bm2vda1.cpp -o /var/folders/j_/zm2rmlh549733gxckfkzj9000000gn/T/tmps3rk0v3a/var/folders/j_/zm2rmlh549733gxckfkzj9000000gn/T/tmp4bm2vda1.o -std=c++11 -fno-math-errno -w" failed with exit status 1
```

#### Reference
https://github.com/serge-sans-paille/pythran/issues/1064: This issue implements a similar function as mine, and I got the idea to rewrite `all_values.sort(kind='mergesort')` to `all_values = np.sort(all_values, kind='mergesort')` through comparing this issue's implementations with mine when debugging.

--- a/pythran/pythonic/__dispatch__/sort.hpp
+++ b/pythran/pythonic/__dispatch__/sort.hpp
@@ -1,0 +1,38 @@
+#ifndef PYTHONIC_DISPATCH_SORT_HPP
+#define PYTHONIC_DISPATCH_SORT_HPP
+
+#include "pythonic/include/__dispatch__/sort.hpp"
+
+#include "pythonic/builtins/list/sort.hpp"
+#include "pythonic/numpy/sort.hpp"
+
+PYTHONIC_NS_BEGIN
+
+namespace __dispatch__
+{
+
+  template <class T, class... Args>
+  auto sort(types::list<T> &l, Args &&... args)
+      -> decltype(pythonic::builtins::list::sort(l,
+                                                 std::forward<Args>(args)...))
+  {
+    return pythonic::builtins::list::sort(l, std::forward<Args>(args)...);
+  }
+  template <class T, class... Args>
+  auto sort(types::list<T> &&l, Args &&... args)
+      -> decltype(pythonic::builtins::list::sort(std::move(l),
+                                                 std::forward<Args>(args)...))
+  {
+    return pythonic::builtins::list::sort(std::move(l),
+                                          std::forward<Args>(args)...);
+  }
+  template <class Any, class... Args>
+  types::none_type sort(Any &&any, Args &&... args)
+  {
+    return pythonic::numpy::ndarray::sort(std::forward<Any>(any),
+                                          std::forward<Args>(args)...);
+  }
+}
+PYTHONIC_NS_END
+
+#endif

--- a/pythran/pythonic/include/__dispatch__/sort.hpp
+++ b/pythran/pythonic/include/__dispatch__/sort.hpp
@@ -1,0 +1,27 @@
+#ifndef PYTHONIC_INCLUDE_DISPATCH_SORT_HPP
+#define PYTHONIC_INCLUDE_DISPATCH_SORT_HPP
+
+#include "pythonic/include/builtins/list/sort.hpp"
+#include "pythonic/include/numpy/sort.hpp"
+
+PYTHONIC_NS_BEGIN
+
+namespace __dispatch__
+{
+
+  template <class T, class... Args>
+  auto sort(types::list<T> &l, Args &&... args)
+      -> decltype(pythonic::builtins::list::sort(l,
+                                                 std::forward<Args>(args)...));
+  template <class T, class... Args>
+  auto sort(types::list<T> &&l, Args &&... args)
+      -> decltype(pythonic::builtins::list::sort(std::move(l),
+                                                 std::forward<Args>(args)...));
+  template <class Any, class... Args>
+  types::none_type sort(Any &&any, Args &&... args);
+
+  DEFINE_FUNCTOR(pythonic::__dispatch__, sort);
+}
+PYTHONIC_NS_END
+
+#endif

--- a/pythran/pythonic/include/numpy/ndarray/sort.hpp
+++ b/pythran/pythonic/include/numpy/ndarray/sort.hpp
@@ -1,0 +1,26 @@
+#ifndef PYTHONIC_INCLUDE_NUMPY_NDARRAY_SORT_HPP
+#define PYTHONIC_INCLUDE_NUMPY_NDARRAY_SORT_HPP
+
+#include "pythonic/include/numpy/sort.hpp"
+
+PYTHONIC_NS_BEGIN
+
+namespace numpy
+{
+  namespace ndarray
+  {
+    template <class E>
+    types::none_type sort(E &&expr, types::none_type);
+
+    template <class E>
+    types::none_type sort(E &&expr, long axis, types::none_type = {});
+
+    template <class E>
+    types::none_type sort(E &&expr, long axis, types::str const &kind);
+
+    DEFINE_FUNCTOR(pythonic::numpy::ndarray, sort);
+  }
+}
+PYTHONIC_NS_END
+
+#endif

--- a/pythran/pythonic/numpy/median.hpp
+++ b/pythran/pythonic/numpy/median.hpp
@@ -40,14 +40,15 @@ namespace numpy
         }
         if (buffer_size % 2 == 1) {
           std::nth_element(buffer.get(), buffer.get() + buffer_size / 2,
-                           buffer_iter, comparator<T>{});
+                           buffer_iter, ndarray::comparator<T>{});
           *out++ = buffer[buffer_size / 2];
         } else {
           std::nth_element(buffer.get(), buffer.get() + buffer_size / 2,
-                           buffer_iter, comparator<T>{});
+                           buffer_iter, ndarray::comparator<T>{});
           auto t0 = buffer[buffer_size / 2];
           std::nth_element(buffer.get(), buffer.get() + buffer_size / 2 - 1,
-                           buffer.get() + buffer_size / 2, comparator<T>{});
+                           buffer.get() + buffer_size / 2,
+                           ndarray::comparator<T>{});
           auto t1 = buffer[buffer_size / 2 - 1];
           *out++ = (t0 + t1) / double(2);
         }
@@ -68,13 +69,13 @@ namespace numpy
     std::unique_ptr<T[]> tmp{new T[n]};
     std::copy(arr.buffer, arr.buffer + n, tmp.get());
     std::nth_element(tmp.get(), tmp.get() + n / 2, tmp.get() + n,
-                     comparator<T>{});
+                     ndarray::comparator<T>{});
     T t0 = tmp[n / 2];
     if (n % 2 == 1) {
       return t0;
     } else {
       std::nth_element(tmp.get(), tmp.get() + n / 2 - 1, tmp.get() + n / 2,
-                       comparator<T>{});
+                       ndarray::comparator<T>{});
       T t1 = tmp[n / 2 - 1];
       return (t0 + t1) / 2.;
     }

--- a/pythran/pythonic/numpy/ndarray/sort.hpp
+++ b/pythran/pythonic/numpy/ndarray/sort.hpp
@@ -1,0 +1,153 @@
+#ifndef PYTHONIC_NUMPY_NDARRAY_SORT_HPP
+#define PYTHONIC_NUMPY_NDARRAY_SORT_HPP
+
+#include "pythonic/include/numpy/ndarray/sort.hpp"
+
+#include <algorithm>
+#include <memory>
+
+#include "pythonic/utils/functor.hpp"
+#include "pythonic/types/ndarray.hpp"
+#include "pythonic/types/str.hpp"
+#include "pythonic/numpy/array.hpp"
+#include "pythonic/utils/pdqsort.hpp"
+
+PYTHONIC_NS_BEGIN
+
+namespace numpy
+{
+  namespace ndarray
+  {
+    namespace
+    {
+      struct quicksorter {
+        template <class... Args>
+        void operator()(Args &&... args)
+        {
+          pdqsort(std::forward<Args>(args)...);
+        }
+      };
+      struct mergesorter {
+        template <class It, class Cmp = std::less<typename It::value_type>>
+        void operator()(It first, It last, Cmp cmp = Cmp())
+        {
+          if (last - first > 1) {
+            It middle = first + (last - first) / 2;
+            operator()(first, middle, cmp);
+            operator()(middle, last, cmp);
+            std::inplace_merge(first, middle, last, cmp);
+          }
+        }
+      };
+      struct heapsorter {
+        template <class... Args>
+        void operator()(Args &&... args)
+        {
+          return std::sort_heap(std::forward<Args>(args)...);
+        }
+      };
+      struct stablesorter {
+        template <class... Args>
+        void operator()(Args &&... args)
+        {
+          return std::stable_sort(std::forward<Args>(args)...);
+        }
+      };
+
+      template <class T>
+      struct _comp;
+      template <class T>
+      struct _comp<std::complex<T>> {
+        bool operator()(std::complex<T> const &i,
+                        std::complex<T> const &j) const
+        {
+          if (std::real(i) == std::real(j))
+            return std::imag(i) < std::imag(j);
+          else
+            return std::real(i) < std::real(j);
+        }
+      };
+
+      template <class T>
+      using comparator =
+          typename std::conditional<types::is_complex<T>::value, _comp<T>,
+                                    std::less<T>>::type;
+
+      template <class T, class pS, class Sorter>
+      typename std::enable_if<std::tuple_size<pS>::value == 1, void>::type
+      _sort(types::ndarray<T, pS> &out, long axis, Sorter sorter)
+      {
+        sorter(out.begin(), out.end(), comparator<T>{});
+      }
+
+      template <class T, class pS, class Sorter>
+      typename std::enable_if<std::tuple_size<pS>::value != 1, void>::type
+      _sort(types::ndarray<T, pS> &out, long axis, Sorter sorter)
+      {
+        constexpr auto N = std::tuple_size<pS>::value;
+        if (axis < 0)
+          axis += N;
+        long const flat_size = out.flat_size();
+        if (axis == N - 1) {
+          const long step = out.template shape<N - 1>();
+          for (T *out_iter = out.buffer, *end_iter = out.buffer + flat_size;
+               out_iter != end_iter; out_iter += step)
+            sorter(out_iter, out_iter + step, comparator<T>{});
+        } else {
+          auto out_shape = sutils::getshape(out);
+          const long step =
+              std::accumulate(out_shape.begin() + axis, out_shape.end(), 1L,
+                              std::multiplies<long>());
+          long const buffer_size = out_shape[axis];
+          const long stepper = step / out_shape[axis];
+          const long n = flat_size / out_shape[axis];
+          long ith = 0, nth = 0;
+          std::unique_ptr<T[]> buffer{new T[buffer_size]};
+          for (long i = 0; i < n; i++) {
+            for (long j = 0; j < buffer_size; ++j)
+              buffer[j] = out.buffer[ith + j * stepper];
+            sorter(buffer.get(), buffer.get() + buffer_size, comparator<T>{});
+            for (long j = 0; j < buffer_size; ++j)
+              out.buffer[ith + j * stepper] = buffer[j];
+
+            ith += step;
+            if (ith >= flat_size) {
+              ith = ++nth;
+            }
+          }
+        }
+      }
+    }
+
+    template <class E>
+    types::none_type sort(E &&expr, long axis, types::none_type)
+    {
+      _sort(expr, axis, quicksorter());
+      return {};
+    }
+
+    template <class E>
+    types::none_type sort(E &&expr, types::none_type)
+    {
+      _sort(expr, 0, quicksorter());
+      return {};
+    }
+
+    template <class E>
+    types::none_type sort(E &&expr, long axis, types::str const &kind)
+    {
+      if (kind == "quicksort")
+        _sort(expr, axis, quicksorter());
+      else if (kind == "mergesort")
+        _sort(expr, axis, mergesorter());
+      else if (kind == "heapsort")
+        _sort(expr, axis, heapsorter());
+      else if (kind == "stable")
+        _sort(expr, axis, stablesorter());
+      return {};
+    }
+  }
+}
+PYTHONIC_NS_END
+
+#endif

--- a/pythran/pythonic/numpy/sort.hpp
+++ b/pythran/pythonic/numpy/sort.hpp
@@ -2,124 +2,18 @@
 #define PYTHONIC_NUMPY_SORT_HPP
 
 #include "pythonic/include/numpy/sort.hpp"
-
-#include <algorithm>
-#include <memory>
-
-#include "pythonic/utils/functor.hpp"
-#include "pythonic/types/ndarray.hpp"
-#include "pythonic/types/str.hpp"
-#include "pythonic/numpy/array.hpp"
-#include "pythonic/utils/pdqsort.hpp"
+#include "pythonic/numpy/ndarray/sort.hpp"
 
 PYTHONIC_NS_BEGIN
 namespace numpy
 {
-  namespace
-  {
-    struct quicksorter {
-      template <class... Args>
-      void operator()(Args &&... args)
-      {
-        pdqsort(std::forward<Args>(args)...);
-      }
-    };
-    struct mergesorter {
-      template <class It, class Cmp = std::less<typename It::value_type>>
-      void operator()(It first, It last, Cmp cmp = Cmp())
-      {
-        if (last - first > 1) {
-          It middle = first + (last - first) / 2;
-          operator()(first, middle, cmp);
-          operator()(middle, last, cmp);
-          std::inplace_merge(first, middle, last, cmp);
-        }
-      }
-    };
-    struct heapsorter {
-      template <class... Args>
-      void operator()(Args &&... args)
-      {
-        return std::sort_heap(std::forward<Args>(args)...);
-      }
-    };
-    struct stablesorter {
-      template <class... Args>
-      void operator()(Args &&... args)
-      {
-        return std::stable_sort(std::forward<Args>(args)...);
-      }
-    };
-
-    template <class T>
-    struct _comp;
-    template <class T>
-    struct _comp<std::complex<T>> {
-      bool operator()(std::complex<T> const &i, std::complex<T> const &j) const
-      {
-        if (std::real(i) == std::real(j))
-          return std::imag(i) < std::imag(j);
-        else
-          return std::real(i) < std::real(j);
-      }
-    };
-
-    template <class T>
-    using comparator = typename std::conditional<types::is_complex<T>::value,
-                                                 _comp<T>, std::less<T>>::type;
-
-    template <class T, class pS, class Sorter>
-    typename std::enable_if<std::tuple_size<pS>::value == 1, void>::type
-    _sort(types::ndarray<T, pS> &out, long axis, Sorter sorter)
-    {
-      sorter(out.begin(), out.end(), comparator<T>{});
-    }
-
-    template <class T, class pS, class Sorter>
-    typename std::enable_if<std::tuple_size<pS>::value != 1, void>::type
-    _sort(types::ndarray<T, pS> &out, long axis, Sorter sorter)
-    {
-      constexpr auto N = std::tuple_size<pS>::value;
-      if (axis < 0)
-        axis += N;
-      long const flat_size = out.flat_size();
-      if (axis == N - 1) {
-        const long step = out.template shape<N - 1>();
-        for (T *out_iter = out.buffer, *end_iter = out.buffer + flat_size;
-             out_iter != end_iter; out_iter += step)
-          sorter(out_iter, out_iter + step, comparator<T>{});
-      } else {
-        auto out_shape = sutils::getshape(out);
-        const long step =
-            std::accumulate(out_shape.begin() + axis, out_shape.end(), 1L,
-                            std::multiplies<long>());
-        long const buffer_size = out_shape[axis];
-        const long stepper = step / out_shape[axis];
-        const long n = flat_size / out_shape[axis];
-        long ith = 0, nth = 0;
-        std::unique_ptr<T[]> buffer{new T[buffer_size]};
-        for (long i = 0; i < n; i++) {
-          for (long j = 0; j < buffer_size; ++j)
-            buffer[j] = out.buffer[ith + j * stepper];
-          sorter(buffer.get(), buffer.get() + buffer_size, comparator<T>{});
-          for (long j = 0; j < buffer_size; ++j)
-            out.buffer[ith + j * stepper] = buffer[j];
-
-          ith += step;
-          if (ith >= flat_size) {
-            ith = ++nth;
-          }
-        }
-      }
-    }
-  }
 
   template <class E>
   types::ndarray<typename E::dtype, types::array<long, E::value>>
   sort(E const &expr, long axis)
   {
     auto out = functor::array{}(expr);
-    _sort(out, axis, quicksorter());
+    ndarray::sort(out, axis);
     return out;
   }
 
@@ -128,7 +22,7 @@ namespace numpy
   sort(E const &expr, types::none_type)
   {
     auto out = functor::array{}(expr).flat();
-    _sort(out, 0, quicksorter());
+    ndarray::sort(out, types::none_type{});
     return out;
   }
 
@@ -137,18 +31,9 @@ namespace numpy
   sort(E const &expr, long axis, types::str const &kind)
   {
     auto out = functor::array{}(expr);
-    if (kind == "quicksort")
-      _sort(out, axis, quicksorter());
-    else if (kind == "mergesort")
-      _sort(out, axis, mergesorter());
-    else if (kind == "heapsort")
-      _sort(out, axis, heapsorter());
-    else if (kind == "stable")
-      _sort(out, axis, stablesorter());
+    ndarray::sort(out, axis, kind);
     return out;
   }
-
-  NUMPY_EXPR_TO_NDARRAY0_IMPL(sort);
 }
 PYTHONIC_NS_END
 

--- a/pythran/tables.py
+++ b/pythran/tables.py
@@ -757,6 +757,10 @@ CLASSES = {
         ),
         "size": AttributeIntr(signature=Fun[[NDArray[T0, :]], int],
                               return_range=interval.positive_values),
+        "sort": MethodIntr(
+            args=("self", "axis", "kind"),
+            defaults=(-1, None)
+        ),
         "strides": AttributeIntr(
             signature=Union[
                 # bool
@@ -4466,6 +4470,7 @@ MODULES = {
         ),
         "pop": MethodIntr(),
         "remove": MethodIntr(),
+        "sort": MethodIntr(),
         "update": MethodIntr(update_effects),
     },
 }
@@ -4575,6 +4580,7 @@ fill_constants_types((), MODULES)
 # a method name to module binding
 # {method_name : ((full module path), signature)}
 methods = {}
+duplicated_methods = {}
 
 
 def save_method(elements, module_path):
@@ -4587,6 +4593,10 @@ def save_method(elements, module_path):
         elif signature.ismethod():
             # in case of duplicates, there must be a __dispatch__ record
             # and it is the only recorded one
+            if elem in MODULES['__dispatch__'] and module_path[0] != '__dispatch__':
+                duplicated_methods.setdefault(elem, []).append((module_path,
+                                                                signature))
+
             if elem in methods and module_path[0] != '__dispatch__':
                 assert elem in MODULES['__dispatch__']
                 path = ('__dispatch__',)

--- a/pythran/tests/test_numpy_func0.py
+++ b/pythran/tests/test_numpy_func0.py
@@ -700,6 +700,9 @@ def np_rosen_der(x):
     def test_sort10(self):
         self.run_test("def np_sort10(a): from numpy import sort ; return sort(3*a, 0)", numpy.arange(2*3*4, 0, -1).reshape(2,3,4), np_sort10=[NDArray[int, :, :, :]])
 
+    def test_sort11(self):
+        self.run_test("def np_sort11(a): a.sort(kind='heapsort'); return a", numpy.arange(2*3*4, 0, -1).reshape(2,3,4), np_sort11=[NDArray[int, :, :, :]])
+
     def test_sort_complex0(self):
         self.run_test("def np_sort_complex0(a): from numpy import sort_complex ; return sort_complex(a)", numpy.array([[1,6],[7,5]]), np_sort_complex0=[NDArray[int,:,:]])
 

--- a/pythran/transformations/normalize_method_calls.py
+++ b/pythran/transformations/normalize_method_calls.py
@@ -4,6 +4,7 @@ from pythran.analyses import Globals, Ancestors
 from pythran.passmanager import Transformation
 from pythran.syntax import PythranSyntaxError
 from pythran.tables import attributes, functions, methods, MODULES
+from pythran.tables import duplicated_methods
 from pythran.conversion import mangle, demangle
 from pythran.utils import isstr
 
@@ -107,8 +108,26 @@ class NormalizeMethodCalls(Transformation):
         else:
             return obj
 
-    def attr_to_func(self, node):
-        mod = methods[node.attr][0]
+    def keyword_based_disambiguification(self, node):
+        assert isinstance(node.func, ast.Attribute)
+        if getattr(node.func.value, 'id', None) != mangle('__dispatch__'):
+            return
+        if not node.keywords:
+            return
+        if node.func.attr not in duplicated_methods:
+            return
+
+        node_keywords = {kw.arg for kw in node.keywords}
+        for disamb_path, disamb_node in duplicated_methods[node.func.attr]:
+            disamb_args = {arg.id for arg in disamb_node.args.args}
+            if all(kw in disamb_args for kw in node_keywords):
+                node.func = self.attr_to_func(node.func, disamb_path)
+                return
+
+
+    def attr_to_func(self, node, mod=None):
+        if mod is None:
+            mod = methods[node.attr][0]
         # Submodules import full module
         self.to_import.add(mangle(mod[0]))
         func = reduce(
@@ -216,6 +235,7 @@ class NormalizeMethodCalls(Transformation):
         >> builtins.__dict__.fromkeys([1, 2, 3])
         """
         node = self.generic_visit(node)
+
         # Only attributes function can be Pythonic and should be normalized
         if isinstance(node.func, ast.Attribute):
             if node.func.attr in methods:
@@ -259,6 +279,7 @@ class NormalizeMethodCalls(Transformation):
                 # Rename module path to avoid naming issue.
                 node.func.value, _ = rec(node.func.value, MODULES)
                 self.update = True
+            self.keyword_based_disambiguification(node)
 
         return node
 

